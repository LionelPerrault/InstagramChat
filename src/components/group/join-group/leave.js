import React, { Fragment } from 'react'
import { leaveGroup } from '../../../utils/group-utils'
import { shape, number, func } from 'prop-types'
import { connect } from 'react-redux'
import PrimaryButton from '../../others/button/primary-btn'

/**
 * Provide dispatch when updateGroups=true
 */

const Leave = ({ leaveDetails, leaved, updateGroups, dispatch }) => {
  let { user, group_id } = leaveDetails

  let leave = e => {
    e.preventDefault()
    leaveGroup({
      user,
      group: group_id,
      updateGroups,
      dispatch,
      done: () => leaved(),
    })
  }

  return (
    <Fragment>
      <PrimaryButton
        label="Leave group"
        onClick={leave}
        extraClass="unfollow"
      />
    </Fragment>
  )
}

Leave.defaultProps = {
  updateGroups: false,
}

Leave.propTypes = {
  leaveDetails: shape({
    user: number.isRequired,
    group_id: number.isRequired,
  }).isRequired,
  leaved: func.isRequired,
}

export default connect()(Leave)
export { Leave as PureLeave }
