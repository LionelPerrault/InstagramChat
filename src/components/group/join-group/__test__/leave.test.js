import React from 'react'
import { create } from 'react-test-renderer'
import { PureLeave } from '../leave'
import { shallow } from 'enzyme'

describe('Leave Component', () => {
  const props = {
    leaveDetails: {
      user: 7,
      group_id: 45,
    },
    leaved: jest.fn(),
  }

  it('should match snapshot', () => {
    const tree = create(<PureLeave {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should should mock leave action when button is clicked', () => {
    const wrapper = shallow(<PureLeave {...props} />)
    wrapper.find('PrimaryButton').simulate('click', {
      preventDefault() {},
    })
  })
})
