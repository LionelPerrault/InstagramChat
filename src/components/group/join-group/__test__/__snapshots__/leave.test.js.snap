// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leave Component should match snapshot 1`] = `
<a
  className="pri_btn unfollow"
  href="#"
  onClick={[Function]}
>
  Leave group
</a>
`;
